src/mem/physical.cc

```diff
@@ -463,21 +463,21 @@ PhysicalMemory::unserializeStore(CheckpointIn &cp)
               range_size, range.size());
 
     uint64_t curr_size = 0;
-    long* temp_page = new long[chunk_size];
-    long* pmem_current;
+    int64_t* temp_page = new int64_t[chunk_size];
+    int64_t* pmem_current;
     uint32_t bytes_read;
     while (curr_size < range.size()) {
         bytes_read = gzread(compressed_mem, temp_page, chunk_size);
         if (bytes_read == 0)
             break;
 
-        assert(bytes_read % sizeof(long) == 0);
+        assert(bytes_read % sizeof(int64_t) == 0);
 
-        for (uint32_t x = 0; x < bytes_read / sizeof(long); x++) {
+        for (uint32_t x = 0; x < bytes_read / sizeof(int64_t); x++) {
```

Comment:
Same here.

I would say uint64_t makes the list sense, but I don't think it matters.

Reply:
See Giacomo's comment above. It was a typo since I was on my phone.